ui: Add Svelte UI scaffolding

juliaogris opened the pull request:
Add Svelte boilerplate for a default svelte app, formatted with prettier
and added to the CI and Docker build, and some node modules used for
development (sirv in particular). Svelte puts static assets in
`frontend/public`, so change the backend to serve that directory instead
of `static/`.

Preview: https://pr24.foxtrot.jul.run

camh- commented on frontend/.gitignore:
```diff
@@ -0,0 +1,4 @@
+/node_modules/
+/public/build/
+
+.DS_Store
```
I assume this was generated which is why it is here, but it is not later removed. At the very least, this does not belong in a subdirectory gitignore file, but I think we agree it does not belong in a repo's gitignore at all.

juliaogris replied:
I have added a bigger purge to the follow-up PR. If you want I could add it here.:
https://github.com/foxygoat/foxtrot/pull/25/commits/973beb15b65f1896ee8d4292c000fab190721b13

juliaogris replied:
ok, I've moved the commit to this PR.

camh- commented on .dockerignore:
```diff
@@ -0,0 +1,3 @@
+/frontend/node_modules/
+/frontend/bundle.*
```
We're not using this path pattern yet in the build. I think it comes later.

juliaogris replied:
yes sorry!

juliaogris replied:
Done.

camh- reviewed (commented):
Just a couple of little things...

juliaogris committed:
ui: Initialise svelte app

Run `npx degit sveltejs/template frontend` installing svelte template at
https://github.com/sveltejs/template/commit/ebde8634dc14732a83df91f789886d1dd2ceed33

juliaogris committed:
ui: Add package-lock.json file for reproducibility

Add package-lock.json file for reproducible builds. To generate
package-lock.json run

	npm install

juliaogris committed:
ui: Remove Typescript setup script

Remove Typescript setup script from sveltejs template is we intend to
work with vanilla JS.

juliaogris committed:
ui: Update generated package.json

Update generated package.json to use foxtrot name instead of svelte-app
and place sirv dependency into dev-dependencies as we use our custom go
webserver to serve built code.

juliaogris committed:
ui: Format with prettier

Run

	npm i --save-dev prettier-plugin-svelte prettier
	npm run format

Copy .prettierc.js from past experience.

juliaogris committed:
ui: Purge unused svelte boilerplate

Purge unused svelte boilerplate from frontend/:

	- README.md
	- .gitignore
	- global.css
	- favicon.png

The README.md and .gitignore at the repo root now cover frontend/ as
well. In a follow up commit we will use TailwindCSS for styles so there
is no need for a global style sheet.

Update index.html to use Foxtrot as the app Title and render a fox emoji
as favicon.

juliaogris committed:
docs: Update README with frontend section

juliaogris committed:
docker: Add frontend build to Docker image

Add frontend build to Dockerfile ignoring generated files such as
node_modules. Update main.go to serve frontend/public directory instead
of old static directory.

juliaogris committed:
github: Add frontend CI

Add frontend CI to github action cicd.yaml and make the release job
depend on it. Run `check-format` and `build` step as part of frontend
CI.

Remove out of place comment on golangci-lint run.

camh- reviewed (approved):
LGTM :herb:
